packages/formation-react/src/components/Telephone/Telephone.jsx

```diff
@@ -0,0 +1,191 @@
+  // return rounded number as a grouped value, e.g. `800` or `1000`
+  // but split non-round numbers, e.g. `827` => `8 2 7`
+  const isRoundNumber = parseInt(number.slice(-len), 10) === 0;
+  return isRoundNumber ? number : number.split('').join(' ');
+};
+
+/**
+ * Format telephone number for label
+ * @param {string} number - Expected a phone number with or without dashes that
+ * matches the number of "#" within the default or set pattern
+ * @return {string} - Combined phone number parts within the label separated by
+ * periods, e.g. "800-555-1212" becomes "800. 5 5 5. 1 2 1 2"
+ */
+const formatTelLabel = number =>
+  number
+    .split(/[^\d]+/)
+    .filter(n => n)
+    .map(formatTelLabelBlock)
+    .join('. ');
+
+/**
+ * Telephone component
+ * @param {string|number} contact (required) - telephone number, with or without
+ *  formatting; all non-digit characters will be stripped out
+ * @param {string} extension (optional) - telephone extension
+ * @param {string} className (optional) - additional space-separated class names
+ *  to add to the link
+ * @param {string} pattern (optional) - Link text format pattern, using "#" as
+ *  the digit placeholder
+ * @param {string} ariaLabel (optional) - if included, this custom aria-label
+ *  will replace the generated aria-label
+ * @param {function} onClick (optional) - function called when the link is
+ *  clicked
+ * @param {string|JSX} children (optional) - if included, this custom
+ *  telephone link text will replace the generated text; the pattern passed in
+ *  will be ignored
+ */
+function Telephone({
+  // phone number (length _must_ match the pattern; leading "1" is removed)
+  contact = '', // telephone number
+  extension = '', // phone extension
+  className = '', // additional css class to add
+  pattern = '', // output format; defaults to patterns.default value
+  ariaLabel = '', // custom aria-label
+  onClick = () => {},
+  children,
+}) {
+  // strip out non-digits for use in href: "###-### ####" => "##########"
+  const parsedNumber = parseNumber(contact.toString());
+  const phoneNumber = CONTACTS[parsedNumber] || parsedNumber;
+
+  // Capture "911" pattern here
+  const contactPattern = pattern || PATTERNS[parsedNumber] || PATTERNS.DEFAULT;
+  const patternLength = contactPattern.match(/#/g).length;
+  const formattedNumber = formatTelText(phoneNumber, contactPattern, extension);
+
+  if (!phoneNumber || phoneNumber.length !== patternLength) {
+    throw new Error(
+      `Contact number "${phoneNumber}" does not match the pattern (${contactPattern})`,
+    );
+  }
+
+  const formattedAriaLabel =
+    ariaLabel ||
+    `${formatTelLabel(formattedNumber)}${
+      extension ? `. extension ${formatTelLabelBlock(extension)}.` : '.'
+    }`;
+
+  const href = `tel:${
+    phoneNumber.length === 10 ? `+1${phoneNumber}` : phoneNumber
+  }${
+    // extension format from RFC3966 https://tools.ietf.org/html/rfc3966#page-5
+    extension ? `;ext=${extension}` : ''
```

Comment:
This [SO answer](https://stackoverflow.com/a/10802815) seems to imply that `;ext=` should be replaced with a comma. I'll stick with the RFC recommendation until I find out otherwise.

Reply:
And [this reference](https://www.howtogeek.com/263546/how-to-dial-phone-extensions-automatically-with-your-iphone/) states that using a comma will "pause" for 2 seconds, while a semi-colon will "wait" until you press a button and dial the remaining numbers.